src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -47,6 +47,10 @@ public function __invoke(Node $node, CompilerContext $context)
             return $this->compileBinaryOperator($node, $context);
         }
 
+        if ($node instanceof Node\Scalar\MagicConst\Dir) {
+            return '__DIR__';
```

Comment:
I realize this doesn't provide a valid case. It just works on my machine.

How would a test for that look like to comply with current test set?